Sofa/framework/Core/test/objectmodel/BaseClass_test.cpp

```diff
@@ -227,6 +233,11 @@ TEST_F(BaseClass_test, checkStaticDefaultTemplate  )
 
     EXPECT_EQ(m_ptr9.getClassName(),"DefaultTemplate3") ;
     EXPECT_EQ(m_ptr9.getTemplateName(),"One,Two,NotAType") ;
+
+    EXPECT_EQ(m_ptr12.getTemplateName(),"f") ;
+    EXPECT_EQ(m_ptr13.getTemplateName(),"vector<f>") ;
+    EXPECT_EQ(m_ptr14.getTemplateName(),"Vec3d") ;
+    EXPECT_EQ(m_ptr15.getTemplateName(),"Vec3d") ;
```

Comment:
This is because Vec3Types and Rigid3Type are imposing their own name with the ::Name() method. 
Have no clear view on that yet... but when I think in term of UX of scene writing, hiding implementation details (and being template on c++ to Vec3Type or to Vec3 is such a detail), then it is perfectly ok to do so. 

I think this is the choice that was done, and this is why there is this ::Name() static method, this was probably because it was considered that doing 
```xml
<MechanicalObject template='Vec3Type'/>"/>
```
instead of
```xml
<MechanicalObject template="Vec3"/>
```
Was adding noise to the user for no gain. 

That's said, the way we are implementing that in our code base can probably be clarified a lot so that we make clear that this is the UX name. Currently in our code base there is always an ambiguity (eg: template in sofa, template in scene) which makes it is fuzzy if the feature is a kind of "introspection/reflexion" mecanism (the kind of stuff that please a lot developpers for doing high quality printf debugging)  or of it is part of the UX.  

This is a problem often encountered when developers are their own/first user.